File: 3236-smallest-missing-integer-greater-than-sequential-prefix-sum/smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution:
    def missingInteger(self, nums: List[int]) -> int:
        pref=nums[0]
        
        for i in range(1,len(nums)):
            if nums[i]!=nums[i-1]+1:
                break

            pref+=nums[i]

        for i in range(pref,52):
            if i not in nums:
                return i

        return pref

File: 3236-smallest-missing-integer-greater-than-sequential-prefix-sum/smallest-missing-integer-greater-than-sequential-prefix-sum.cpp
class Solution {
public:
    int missingInteger(vector<int>& nums) {
        int pref=nums[0];
        
        for (int i=1; i<nums.size(); i++){
            if (nums[i]!=nums[i-1]+1)
                break;

            pref+=nums[i];
        }

        std::unordered_set<int> seen(nums.begin(), nums.end());

        while (seen.count(pref)) {
            pref++;
        }

        return pref;
    }
};